cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go

```diff
@@ -874,7 +875,21 @@ func TestNodeGroupDeleteNodesTwice(t *testing.T) {
 		// finalizers.
 		now := v1.Now()
 		machine.DeletionTimestamp = &now
-		return controller.machineInformer.Informer().GetStore().Update(newUnstructuredFromMachine(machine))
+		if _, err := controller.dynamicclient.Resource(*controller.machineResource).Namespace(machine.GetNamespace()).Update(context.Background(), newUnstructuredFromMachine(machine), v1.UpdateOptions{}); err != nil {
+			return err
+		}
+
+		// Ensure the update worked
```

Comment:
curious if there's any particular reason you saw to add this extra check? seems fair to trust the error returned in 879.

Reply:
This was actually to make sure the update had propagated and the client that is used by the nodegroup could read the deletiontimestamp, I'll update the comment, I'll also try seeing if this is as stable if I remove this.

I had used stress before to try and get it to fail but it didn't, I can retry that